index_impl.go

```diff
@@ -482,7 +483,16 @@ func (i *indexImpl) SearchInContext(ctx context.Context, req *SearchRequest) (sr
 	ctx = context.WithValue(ctx, search.SearchIOStatsCallbackKey,
 		search.SearchIOStatsCallbackFunc(sendBytesRead))
 
-	searcher, err := req.Query.Searcher(ctx, indexReader, i.m, search.SearcherOptions{
+	// Using a conjunction query to get intersection of results from similarity query
+	// and the original query
+	searchQuery := req.Query
+	if req.Similarity != nil {
+		similarityQuery := query.NewSimilarityQuery(req.Similarity.Vector)
+		similarityQuery.SetFieldVal(req.Similarity.Field)
```

Comment:
I'm thinking if it makes sense to have this as a set of fields instead, and what you're trying to do is apply the similarity search over each of those fields. So, you'd be getting the similar vectors from all these indexed fields and do a conjunction of these sub-similarity searchers and the user's non-similarity query.

Reply:
As per discussion, the user can enter multiple fields, not just one.

Reply:
@abhinavdangeti do you think it's a valid use case where the user enters multiple fields against 1 vector?
@Thejas-bhat mentioned that a user could have multiple fields with dense vector content and might want to query the same vector across them and hence, the need for multiple fields.